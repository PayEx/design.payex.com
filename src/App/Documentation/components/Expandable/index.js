import React, { useEffect } from "react";

import { ComponentPreview, DocContainer } from "#";
import expandable from "$/px-script/main/expandable";
import ExpandableComponent from "@/Expandable";

const items = [
    {
        title: "Foo",
        content: <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer tristique scelerisque blandit. Vestibulum venenatis elit non nunc pulvinar, vitae tincidunt quam malesuada.
            Etiam orci dui, ultricies consectetur venenatis sit amet, ullamcorper quis metus. Aliquam eget consequat erat. Praesent arcu tortor, eleifend ac lorem eu, ullamcorper tristique purus.
            Phasellus felis mi, malesuada et egestas eget, efficitur at magna. Fusce euismod, elit tempor vulputate mollis, diam eros venenatis libero, eget ultricies diam velit at erat.
            Morbi mattis scelerisque augue, ut volutpat lectus sodales iaculis.
        </p>
    },
    {
        title: "Bar",
        content: <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer tristique scelerisque blandit. Vestibulum venenatis elit non nunc pulvinar, vitae tincidunt quam malesuada.
            Etiam orci dui, ultricies consectetur venenatis sit amet, ullamcorper quis metus. Aliquam eget consequat erat. Praesent arcu tortor, eleifend ac lorem eu, ullamcorper tristique purus.
            Phasellus felis mi, malesuada et egestas eget, efficitur at magna. Fusce euismod, elit tempor vulputate mollis, diam eros venenatis libero, eget ultricies diam velit at erat.
            Morbi mattis scelerisque augue, ut volutpat lectus sodales iaculis.
        </p>
    },
    {
        title: "Baz",
        content: <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer tristique scelerisque blandit. Vestibulum venenatis elit non nunc pulvinar, vitae tincidunt quam malesuada.
            Etiam orci dui, ultricies consectetur venenatis sit amet, ullamcorper quis metus. Aliquam eget consequat erat. Praesent arcu tortor, eleifend ac lorem eu, ullamcorper tristique purus.
            Phasellus felis mi, malesuada et egestas eget, efficitur at magna. Fusce euismod, elit tempor vulputate mollis, diam eros venenatis libero, eget ultricies diam velit at erat.
            Morbi mattis scelerisque augue, ut volutpat lectus sodales iaculis.
        </p>
    }
];

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Accordion.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ExpandableComponent items={items} />
        </ComponentPreview>
    </>
);

const Accordion = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Accordion.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="accordion">
                <ExpandableComponent items={items} />
            </div>
        </ComponentPreview>
    </>
);

const Expandable = () => {
    useEffect(() => expandable.init());

    return (
        <DocContainer docToc>
            <p className="lead">
                Accordion
            </p>
            <Overview />
            <Accordion />
        </DocContainer>
    );
};

export default Expandable;

/* For testing */
export { Overview, Accordion };
