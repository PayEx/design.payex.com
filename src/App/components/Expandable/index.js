import React from "react";

/* The expandable component acceps either an array or a single object [AW] */
const Expandable = ({ items }) => (
    items ? (Array.isArray(items) ?
        items.map(({ title, content }, i) => (
            <div key={i} className="expandable">
                <div className="expandable-header">
                    <h5>{title}</h5>
                </div>
                <div className="expandable-content">
                    {content}
                </div>
            </div>
        )) :
        <div className="expandable">
            <div className="expandable-header">
                <h5>{items.title}</h5>
            </div>
            <div className="expandable-content">
                {items.content}
            </div>
        </div>
    ) : null
);

export default Expandable;
