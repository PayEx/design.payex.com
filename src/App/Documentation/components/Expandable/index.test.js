import React from "react";
import { shallow } from "enzyme";

import Expandable, { Overview, Accordion } from "./index";

describe("Components: Expandable", () => {
    it("is defined", () => {
        expect(Expandable).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Expandable />);

        expect(wrapper).toMatchSnapshot();
    });

    // describe("Overview", () => {
    //     it("is defined", () => {
    //         expect(Overview).toBeDefined();
    //     });

    //     it("renders", () => {
    //         const wrapper = shallow(<Overview />);

    //         expect(wrapper).toMatchSnapshot();
    //     });
    // });

    // describe("Accordion", () => {
    //     it("is defined", () => {
    //         expect(Accordion).toBeDefined();
    //     });

    //     it("renders", () => {
    //         const wrapper = shallow(<Accordion />);

    //         expect(wrapper).toMatchSnapshot();
    //     });
    // });

    test.todo("Expandable tests");
});
