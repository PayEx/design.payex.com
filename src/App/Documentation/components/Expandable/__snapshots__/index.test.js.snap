// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Expandable renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Accordion
  </p>
  <Overview />
  <Accordion />
</DocContainer>
`;
